pages/plan/index.js

```diff
@@ -65,27 +67,35 @@ export default function Plan({
     .filter((recipe) => recipe.hasCooked)
     .map((recipe) => recipe.recipe);
 
+  function getCalendarDayFromDb(date) {
+    return user.calendar.find((calendarDay) => calendarDay.date === date);
```

Comment:
```suggestion
    return user.calendar.findOne((calendarDay) => calendarDay.date === date);
```
Maybe its better to use findOne to make more clear that you are only looking for the matching date :)

Reply:
I think findOne does not exist as an array method!
Maybe you were thinking of a method for MongoDB?